apps-rendering/src/client/newsletterEmbeds.ts

```diff
@@ -0,0 +1,61 @@
+type NewsletterHeightEventType = { source: MessageEventSource; origin: string };
+type MessageType = {
+	type: string;
+	value: string;
+};
+
+const allowedOrigins = ['https://www.theguardian.com'];
+export default (): Promise<void> => {
+	const allIframes: HTMLIFrameElement[] = [].slice.call(
+		document.querySelectorAll('.email-sub__iframe'),
+	);
+
+	// Tell the iframes to resize once this script is loaded
+	// Otherwise, earlier resize events might be missed
+	// So we don't have to load this script as a priority on each load
+	allIframes.forEach((iframe) => {
+		iframe.contentWindow?.postMessage('resize', '*');
+	});
+
+	window.addEventListener('message', (event) => {
+		if (
+			!allowedOrigins.includes(
+				(event as NewsletterHeightEventType).origin,
+			)
+		)
+			return;
+
+		const iframes: HTMLIFrameElement[] = allIframes.filter((i) => {
+			try {
+				return (
+					i.contentWindow ===
+					(event as NewsletterHeightEventType).source
```

Comment:
TypeScript says that it's possible for both `contentWindow` and `source` to be `null`. In that scenario I think this check will return `true` and include the `iframe` in the filter, which I'm guessing isn't what you intend here? Would it be best to include a check against these properties being `null`?

I believe this check is also being used in the `mobile-apps-article-templates` implementation, so it may also be a good idea to include the `null` check there too?